pkg/archive/archive.go

```diff
@@ -1218,16 +1219,27 @@ func cmdStream(cmd *exec.Cmd, input io.Reader) (io.ReadCloser, error) {
 		return nil, err
 	}
 
+	// Ensure the command has exited before we clean anything up
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
@stbenjam I personally use WaitGroups when there's more than one thing to wait on, otherwise I just close a `chan struct{}`, but no big deal. Otherwise PR looks good to me.

Reply:
Sure, no problem, I suppose that's more efficient. I updated it to use `chan struct{}`.